### ISSUE TITLE
Regression caused by #655

### ISSUE THREAD

**Issue by UebelAndre** [bazelbuild/rules_python#677]
# 🐞 bug report

### Affected Rule

 The issue is caused by the rule: 
`pip_repository` breaks if `BUILD_WORKING_DIRECTORY` is set where the `requirements.bzl` file is not written.

### Is this a regression?

This is a regression caused by https://github.com/bazelbuild/rules_python/pull/655#discussion_r839916855

### Description

 A clear and concise description of the problem...
`pip_repository` should always write a `requirements.bzl` file. Printing the contents should be a sideeffect.

## 🔬 Minimal Reproduction

From the root of `rules_python`:

```shell
cd examples/pip_parse && export BUILD_WORKING_DIRECTORY=$(pwd) && bazel test //...
```

## 🔥 Exception or Error

```
pip_parse % bazel test //...
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 9b05ae6c-b5e9-42f6-9820-376b018e7233
ERROR: error loading package '': cannot load '@pypi//:requirements.bzl': no such file
INFO: Elapsed time: 6.276s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
```

**Comment by UebelAndre** [comment]
cc @alexeagle

**Comment by rahul-theorem** [comment]
I also hit this error specifically when switching branches that had different requirements. We resolved this by downgrading to rules_python 0.7.0, but hopefully a fix for this will make it into the next release.

**Comment by alexeagle** [comment]
Sorry about that, I'm trying to fix in https://github.com/bazelbuild/rules_python/pull/679 but got stuck there
